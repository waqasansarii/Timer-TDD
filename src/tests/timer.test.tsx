import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import * as ReactDOM from 'react-dom'
import Timer from '../component/timer'
import { act } from 'react-dom/test-utils'

test('shuld render timer com', () => {
  const container = render(<Timer />)
  expect(container.getByTestId("timer")).toBeInTheDocument()
})

test('should render heading', () => {
  const heading = render(<Timer />)
  const headingElm = heading.getByRole('heading')
  expect(headingElm).toHaveTextContent('Timer App')
})
test('should render timer initial values hours, mins and seconds 00 ', () => {
  const initial = render(<Timer />)
  const initialValue = initial.getAllByText(/00/i)
  expect(initialValue).toHaveLength(3)
})
test('should render texts', () => {
  const span = render(<Timer />)
  expect(span.getByText(/hours/i && /minutes/i && /seconds/i)).toBeInTheDocument()
})

test('start btn', () => {
  const start = render(<Timer />)
  expect(start.getByText(/start/i)).toBeInTheDocument()

})


it('start and stop onclick', () => {

  const startbtn = render(<Timer />)
  const btnelm = startbtn.getByText(/start/i)
  expect(btnelm).toBeTruthy()
  act(() => {
    fireEvent.click(btnelm)
    
  })
  expect(btnelm.innerHTML).toBe("stop")
  

})





