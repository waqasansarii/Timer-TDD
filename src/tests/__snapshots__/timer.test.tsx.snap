// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reset 1`] = `
<button
  class="stop"
>
  stop
</button>
`;
